dynamics/client.py

```diff
@@ -428,6 +446,7 @@ def post(self, data: Dict[str, Any], *, query: Optional[str] = None) -> Dict[str
             data=json.dumps(data).encode(),
             headers={**self.default_headers("post"), **self.headers},
         )
+        response.raise_for_status()
```

Comment:
See [this comment](https://github.com/MrThearMan/dynamics-client/pull/4/files#r1148326348).